Src/DependencyCollector/Shared/DependencyTrackingTelemetryModule.cs

```diff
@@ -298,5 +303,19 @@ private void InitializeForRuntimeInstrumentationOrFramework()
             }
         }
 #endif
+
+#if !NET40
+        [MethodImpl(MethodImplOptions.NoOptimization | MethodImplOptions.NoInlining)]
+        private void PrepareActivity()
```

Comment:
no, there is no check. If Activity static stuff was already initialized, this method costs nothing, considering it's done once per DependencyTrackingTelemetryModule initialization.